### PR TITLE
Add coverage for VMware NVME controllers

### PR DESCRIPTION
**Pull request by shubhamsg199**
Adding coverage for provisioning VMware with NVME controllers(SAT-23558)

### CODE REVIEW

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_computeresource_vmware.py -k test_positive_provision_end_to_end
provisioning: true
pod_resources_size: extra_extra_large

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_computeresource_vmware.py -k test_positive_provision_end_to_end
provisioning: true
pod_resources_size: extra_extra_large

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7976
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_computeresource_vmware.py -k test_positive_provision_end_to_end --external-logging --include-onprem-provisioning
Test Result : ========= 24 failed, 4 deselected, 514 warnings in 2691.89s (0:44:51) ==========
```

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_end_to_end
provisioning: true
pod_resources_size: extra_extra_large

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7984
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_end_to_end --external-logging --include-onprem-provisioning
Test Result : ================= 8 passed, 326 warnings in 4826.28s (1:20:26) =================
```

**Review by shweta83 — APPROVED** [review]
Ack

File: tests/foreman/api/test_computeresource_vmware.py [hunk 1]
```diff
@@ -79,11 +79,20 @@ def test_positive_provision_end_to_end(
...
-                    'size_gb': 10,
+                    'size_gb': 8,
                     'thin': '1',
                     'storage_pod': f'{settings.vmware.datastore_cluster}',
+                    'controller_key': 2001,
+                },
+                '1': {
+                    'size_gb': 8,
+                    'thin': '1',
+                    'storage_pod': f'{settings.vmware.datastore_cluster}',
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_computeresource_vmware.py:94]: Same for above
```suggestion
                    'storage_pod': settings.vmware.datastore_cluster,
```

File: tests/foreman/api/test_computeresource_vmware.py [hunk 2]
```diff
@@ -79,11 +79,20 @@ def test_positive_provision_end_to_end(
...
             'guest_id': 'rhel8_64Guest',
+            'scsi_controllers': [{"type": "ParaVirtualSCSIController", "key": 1001}],
+            'nvme_controllers': [{"type": "VirtualNVMEController", "key": 2001}],
             'volumes_attributes': {
                 '0': {
-                    'size_gb': 10,
+                    'size_gb': 8,
                     'thin': '1',
                     'storage_pod': f'{settings.vmware.datastore_cluster}',
+                    'controller_key': 2001,
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_computeresource_vmware.py:89]: is this some random key we specifying?

File: tests/foreman/api/test_computeresource_vmware.py [hunk 3]
```diff
@@ -79,11 +79,20 @@ def test_positive_provision_end_to_end(
             'cluster': f'{settings.vmware.cluster}',
             'start': '1',
             'guest_id': 'rhel8_64Guest',
+            'scsi_controllers': [{"type": "ParaVirtualSCSIController", "key": 1001}],
+            'nvme_controllers': [{"type": "VirtualNVMEController", "key": 2001}],
             'volumes_attributes': {
                 '0': {
-                    'size_gb': 10,
+                    'size_gb': 8,
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_computeresource_vmware.py:86]: why specifically 8gb here? and How will OS handle this?

File: tests/foreman/api/test_computeresource_vmware.py [hunk 4]
```diff
@@ -79,11 +79,20 @@ def test_positive_provision_end_to_end(
             'cluster': f'{settings.vmware.cluster}',
             'start': '1',
             'guest_id': 'rhel8_64Guest',
+            'scsi_controllers': [{"type": "ParaVirtualSCSIController", "key": 1001}],
+            'nvme_controllers': [{"type": "VirtualNVMEController", "key": 2001}],
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_computeresource_vmware.py:83]:
```suggestion
            'scsi_controllers': [{'type': 'ParaVirtualSCSIController', 'key': 1001}],
            'nvme_controllers': [{'type': 'VirtualNVMEController', 'key': 2001}],
```

File: tests/foreman/api/test_computeresource_vmware.py (hunk 2 above)
Comment (shubhamsg199) [reply tests/foreman/api/test_computeresource_vmware.py:89]: Yes any random key, but has to be matched with any of the controller specified above

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_end_to_end
provisioning: true
pod_resources_size: extra_extra_large

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8071
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_end_to_end --external-logging --include-onprem-provisioning
Test Result : ================= 8 passed, 317 warnings in 5066.12s (1:24:26) =================
```

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**